client/mysqlslap.c

```diff
@@ -1855,12 +1886,17 @@ pthread_handler_t run_task(void *p)
   ulonglong counter= 0, queries;
   ulonglong detach_counter;
   unsigned int commit_counter;
+  ulong sleep_tm= 0;
+  struct timeval last_tmstamp, tmstamp;
   MYSQL *mysql;
   MYSQL_RES *result;
   MYSQL_ROW row;
   statement *ptr;
   thread_context *con= (thread_context *)p;
 
+  if (unlikely(con->srv_time_target))
```

Comment:
No need for unlikely() attribute here. Also, this needs to go right before the for loop, otherwise you're timing connection delay the first time. Not too big of a deal but might as well make it consistent.